btk/draw_blends.py

```diff
@@ -56,195 +58,102 @@ def get_size(pixel_scale, catalog, i_obs_cond):
-    mini-batch analyzed separately. The results are then combined to output a
-    dict with results of entire batch. If multiprocessing is true, then each of
-    the mini-batches are run in parallel.
-
-    Args:
-        Args: Class containing parameters to create blends
-        blend_generator: Generator to create blended object
-        observing_generator: Creates observing conditions for each entry in
-            batch.
-        multiprocessing: Divides batch of blends to draw into mini-batches and
-            runs each on different core
-        cpus: If multiprocessing, then number of parallel processes to run.
-
-    Yields:
-        Dictionary with blend images, isolated object images, blend catalog,
-        and observing conditions.
-    """
-    while True:
+class DrawBlendsGenerator(ABC):
+    def __init__(
+        self,
+        blend_generator,
+        observing_generator,
+        meas_band="i",
+        multiprocessing=False,
+        cpus=1,
+        verbose=False,
+        add_noise=True,
+        min_snr=0.05,
+    ):
+        """Class that generates images of blended objects, individual isolated
+        objects, for each blend in the batch.
+
+        Batch is divided into mini batches of size blend_generator.batch_size//cpus and
+        each mini-batch analyzed separately. The results are then combined to output a
+        dict with results of entire batch. If multiprocessing is true, then each of
+        the mini-batches are run in parallel.
+
+        Args:
+            blend_generator: Object generator to create blended object
+            observing_generator: Observing generator for each entry in batch.
+            multiprocessing: Divides batch of blends to draw into mini-batches and
+                runs each on different core
+            cpus: If multiprocessing, then number of parallel processes to run.
+            meas_band:
+        """
+
+        self.blend_generator = blend_generator
+        self.observing_generator = observing_generator
+        self.multiprocessing = multiprocessing
+        self.cpus = cpus
+
+        self.batch_size = self.blend_generator.batch_size
+        self.max_number = self.blend_generator.max_number
+        self.stamp_size = self.observing_generator.stamp_size
+        self.pixel_scale = self.observing_generator.obs_conditions.pixel_scale
+        self.bands = self.observing_generator.bands
+        self.meas_band = meas_band
+
+        self.add_noise = add_noise
+        self.min_snr = min_snr
+
+        self.verbose = verbose
+
+    def __iter__(self):
+        return self
+
+    def __next__(self):
+        """
+        Returns:
+            Dictionary with blend images, isolated object images, blend catalog,
+            and observing conditions.
+        """
         batch_blend_cat, batch_obs_cond, batch_wcs = [], [], []
-        stamp_size = np.int(Args.stamp_size / Args.pixel_scale)
+        pix_stamp_size = int(self.stamp_size / self.pixel_scale)
```

Comment:
Hmm now that I think about it, fixing one pixel_scale for the draw_blend_generator might be a problem for multi resolution.